internal/locate/region_cache.go

```diff
@@ -1559,6 +1703,33 @@ func (c *RegionCache) BatchLoadRegionsWithKeyRange(bo *retry.Backoffer, startKey
 	return
 }
 
+// BatchLoadRegionsWithKeyRanges loads at most given numbers of regions to the RegionCache,
+// within the given key range from the key ranges. Returns the loaded regions.
+func (c *RegionCache) BatchLoadRegionsWithKeyRanges(bo *retry.Backoffer, keyRanges []pd.KeyRange, count int, opts ...BatchLocateKeyRangesOpt) (regions []*Region, err error) {
+	if len(keyRanges) == 0 {
+		return nil, nil
+	}
+	regions, err = c.batchScanRegions(bo, keyRanges, count, opts...)
+	if err != nil {
+		return
+	}
+	if len(regions) == 0 {
+		startKey, endKey := keyRanges[0].StartKey, keyRanges[len(keyRanges)-1].EndKey
+		err = errors.Errorf("PD returned no region, first range start_key: %q, last range end_key: %q, encode_start_key: %q, encode_end_key: %q",
+			util.HexRegionKeyStr(startKey), util.HexRegionKeyStr(endKey),
+			util.HexRegionKeyStr(c.codec.EncodeRegionKey(startKey)), util.HexRegionKeyStr(c.codec.EncodeRegionKey(endKey)))
+		return
+	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	for _, region := range regions {
```

Comment:
Do we have a specific `update` interface of the `RegionCache`?

Reply:
No, there are only `UpdateLeader` and `UpdateBucketsIfNeeded`

BTW, I'm not sure if `UpdateRegion` is resonable, because there can be multi intersected regions when you want to update one region, the name `update` looks like only one old region info will be replaced by the new one.